frontend/src/scenes/insights/InsightTooltip/insightTooltipUtils.tsx

```diff
@@ -85,8 +84,9 @@ export function getFormattedDate(dayInput?: string | number, interval?: Interval
 }
 
 export function invertDataSource(seriesData: SeriesDatum[]): InvertedSeriesDatum[] {
-    const { cohorts } = useValues(cohortsModel)
-    const { formatPropertyValueForDisplay } = useValues(propertyDefinitionsModel)
+    // NOTE: These logics should not be loaded as hooks unless converted to a "use" function
```

Comment:
Hack looks good to me conceptually. Though I'd change the warning:

```suggestion
    // NOTE: Assuming these logics are mounted elsewhere, and we're not interested in tracking changes.
```

This will fail if `cohortsModel` is not mounted. The way around it is:

```
const cohorts = cohortsModel.findMounted()?.values?.cohorts`
```

Which doesn't throw, but needs some extra `undefined` handling.

Alternatively, everything can be refactored in a way to preserve hook rendering order, but 🚢